documentation/modules/metrics/con-metrics-cruise-control.adoc

```diff
@@ -0,0 +1,96 @@
+You can use your own configuration or use the example `kafka-cruise-control-metrics.yaml` file provided by Strimzi.
+
+You add the configuration to the `metricsConfig` of the `CruiseControl` property in the `Kafka` resource.
+The configuration enables JMX for Cruise Control and uses the {JMXExporter} to expose Cruise Control metrics through an HTTP endpoint.
+The HTTP endpoint is scraped by the Prometheus server.
+
+.Example metrics configuration for Cruise Control
+[source,yaml,subs="+quotes,attributes"]
+----
+  apiVersion: {KafkaApiVersion}
+  kind: Kafka
+  metadata:
+    name: my-cluster
+  Spec:
+    # ...
+    cruiseControl:
+      # ...
+      metricsConfig:
+        type: jmxPrometheusExporter
+        valueFrom:
+          configMapKeyRef:
+            name: cruise-control-metrics
+            key: metrics-config.yml
+  ---
+  kind: ConfigMap
+  apiVersion: v1
+  metadata:
+    name: cruise-control-metrics
+    labels:
+      app: strimzi
+  data:
+    metrics-config.yml: |
+    # _metrics configuration..._
+----
+
+== Viewing Cruise Control metrics
+
+After you expose the Cruise Control metrics, you can use Prometheus or another suitable monitoring system to view information on the metrics data.
+Strimzi provides an xref:assembly-metrics-config-files-str[example Grafana dashboard] to display visualizations of Cruise Control metrics.
+The dashboard is a JSON file called `strimzi-cruise-control.json`.
+The exposed metrics provide the monitoring data when you xref:proc-metrics-grafana-dashboard-str[enable the Grafana dashboard].
+
+=== Monitoring balancedness scores
+
+Cruise Control metrics include a balancedness score.
+Balancedness is the measure of how evenly a workload is distributed in a Kafka cluster.
+
+The Cruise Control metric for balancedness score (`balancedness-score`) differs from the balancedness score in the `KafkaRebalance` resource.
+Cruise Control calculates each score using different goals.
+The former uses `anomaly.detection.goals`, while the latter uses `default.goals`.
+The goals are specified in the `spec.cruiseControl.config` of the `Kafka` custom resource.
+
+[NOTE]
+====
+Refreshing the `KafkaRebalance` resource fetches an optimization proposal.
+The latest cached optimization proposal is fetched if one of the following conditions applies:
+
+* KafkaRebalance `goals` match the goals configured in the `default.goals`section of the `Kafka` resource
+* KafkaRebalance `goals` are not specified
+
+Otherwise, Cruise Control generates a new optimization proposal based on KafkaRebalance `goals`. If new proposals are generated with each refresh, this can impact performance monitoring.
+====
+
+=== Alerts on anomaly detection
+
+Cruise Control’s anomaly detection is enabled for use with Strimzi.
+You can use anomaly detection when monitoring Cruise Control metrics.
```

Comment:
I don't think this makes sense. It suggests that the user should somehow apply some anomaly detection tool or technique on the CC metrics data. But that contradicts the other parts of the docs. IMHO it should be deleted, or maybe it should be something like _Anomaly detection results are part of the Cruise Control metrics._ But I think it would be good to provide more details of what will be the metric names etc.

Reply:
I think we can lose those first two sentences.
And add another link to see the list of anomaly metrics. (We also have one in the intro)
So we have the following.
Not sure what you mean by adding more details on metric names?

> Cruise Control’s anomaly notifier provides metrics data for conditions that block the generation of optimization goals, such as broker failures, and any actions taken.
> If you want more visibility, you can use these metrics to set up alerts and send out notifications.
> You can use Prometheus to generate alerts.
> Prometheus Alertmanager can then route the alerts through a notification channel.
> 
> The {CruiseControlSensorsDocs} provides information on `AnomalyDetector` metrics.

Reply:
I agree with your suggestions Paul but I think you missed the edits in your example in the above! The first two sentences are still there

Reply:
Thanks, Kyle. I haven't pushed the changes up yet. Waiting on clarification of JMX, which we have now.

Reply:
Updated as follows:

> **Alerts on anomaly detection**
> 
> Cruise control's _anomaly detector_ provides metrics data for conditions that block the generation of optimization goals, such as broker failures.
> If you want more visibility, you can use the metrics provided by the anomaly detector to set up alerts and send out notifications.
> You can set up Cruise Control’s _anomaly notifier_ to route alerts based on these metrics through a specified notification channel.
> Alternatively, you can set up Prometheus to scrape the metrics data provided by the anomaly detector and generate alerts.
> Prometheus Alertmanager can then route the alerts generated by Prometheus.
> 
> The {CruiseControlConfigDocs} provides information on `AnomalyDetector` metrics and the anomaly notifier.